### DISCUSSION TITLE
504 (Gateway Timeout)

### DISCUSSION THREAD

**Discussion by justerenoir** [vercel/community#5957] [Help] [answered]
### Summary

I am running replicate to generate an image of a home interior. The image is generated, fetched and received on my local machine in development but when I run on Vercel it is giving me a 504 error. 

Not sure if vercel server is timing out before I have time to receive the image from replicate. The project is pixelperfecthome.com.

### Steps to Reproduce

upload an image, choose the room type, choose a theme, and generate.

**Comment by amyegan** [comment, marked as answer]
Hey Renoir. A 504 [INTERNAL_EDGE_FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/INTERNAL_EDGE_FUNCTION_INVOCATION_TIMEOUT#) does indicate that the function took too long to return the image. That is likely because it took too long to get the response back from replicate, as you suspect.

If you're on a Pro plan, you can [configure max duration](https://vercel.com/docs/functions/configuring-functions/duration#) to be as high is 300 seconds (5 minutes). Hobby accounts are limited to a maximum of 10 seconds.

Another option is [streaming data](https://vercel.com/docs/functions/streaming#) which can give your function (and replicate) more time to finish processing.
